packages/vite/src/node/plugins/workerImportMetaUrl.ts

```diff
@@ -132,7 +132,10 @@ export function workerImportMetaUrlPlugin(config: ResolvedConfig): Plugin {
         if (s) {
           return {
             code: s.toString(),
-            map: config.build.sourcemap ? s.generateMap({ hires: true }) : null
+            map:
+              !isBuild || config.build.sourcemap
+                ? s.generateMap({ hires: true, source: id })
+                : null
```

Comment:
Should we create a helper that takes the magic string, the id, and the config and does this logic? Maybe `transformResult(s, id, config)`